Update installation instructions and dependencies

jaclark5 opened the pull request:
Partial fix to #6 where instructions to run tests and dependencies were updated.

lmhale99 commented:
The documentation probably needs updating throughout for newer iprPy...

As for the requirements, spglib and phonopy are technically optional requirements that the base package doesn't need but some of the modular calculations do.  I'm hesitant to put them in setup.py at the moment, but I need to think about it some more.

jaclark5 commented:
@lmhale99, I see what you're saying. According to the [python best practices](https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/) I moved those to a "requirements.txt" file. For added measure I added a more meaningful import error for each in their calculation file.
